internal/dag/gatewayapi_processor.go

```diff
@@ -1310,6 +1393,208 @@ func (p *GatewayAPIProcessor) computeHTTPRoute(route *gatewayapi_v1beta1.HTTPRou
+			// Convert method match to path match
+			pathMatch, ok := gatewayGRPCMethodMatchCondition(match.Method, routeAccessor)
+			if !ok {
+				continue
+			}
+
+			headerMatches, err := gatewayGRPCHeaderMatchConditions(match.Headers)
+			if err != nil {
+				routeAccessor.AddCondition(gatewayapi_v1alpha2.RouteConditionAccepted, metav1.ConditionFalse, gatewayapi_v1beta1.RouteReasonUnsupportedValue, err.Error())
+				continue
+			}
+
+			matchconditions = append(matchconditions, &matchConditions{
+				path:    pathMatch,
+				headers: headerMatches,
+			})
+		}
+
+		// Process rule-level filters.
+		var (
+			requestHeaderPolicy, responseHeaderPolicy *HeadersPolicy
+			mirrorPolicy                              *MirrorPolicy
+		)
+
+		for _, filter := range rule.Filters {
+			switch filter.Type {
+			case gatewayapi_v1alpha2.GRPCRouteFilterRequestHeaderModifier:
+				// Per Gateway API docs, "specifying a core filter multiple times has
+				// unspecified or custom conformance.", here we choose to just process
+				// the first one.
+				if requestHeaderPolicy != nil {
+					continue
+				}
+
+				var err error
+				requestHeaderPolicy, err = headersPolicyGatewayAPI(filter.RequestHeaderModifier, string(filter.Type))
+				if err != nil {
+					routeAccessor.AddCondition(gatewayapi_v1beta1.RouteConditionResolvedRefs, metav1.ConditionFalse, status.ReasonDegraded, fmt.Sprintf("%s on request headers", err))
+				}
+			case gatewayapi_v1alpha2.GRPCRouteFilterResponseHeaderModifier:
+				// Per Gateway API docs, "specifying a core filter multiple times has
+				// unspecified or custom conformance.", here we choose to just process
+				// the first one.
+				if responseHeaderPolicy != nil {
+					continue
+				}
+
+				var err error
+				responseHeaderPolicy, err = headersPolicyGatewayAPI(filter.ResponseHeaderModifier, string(filter.Type))
+				if err != nil {
+					routeAccessor.AddCondition(gatewayapi_v1beta1.RouteConditionResolvedRefs, metav1.ConditionFalse, status.ReasonDegraded, fmt.Sprintf("%s on response headers", err))
+				}
+			case gatewayapi_v1alpha2.GRPCRouteFilterRequestMirror:
+				// Get the mirror filter if there is one. If there are more than one
+				// mirror filters, "NotImplemented" condition on the Route is set to
+				// status: True, with the "NotImplemented" reason.
+				if mirrorPolicy != nil {
+					routeAccessor.AddCondition(status.ConditionNotImplemented, metav1.ConditionTrue, status.ReasonNotImplemented, "GRPCRoute.Spec.Rules.Filters: Only one mirror filter is supported.")
+					continue
+				}
+
+				if filter.RequestMirror != nil {
+					mirrorService, cond := p.validateBackendObjectRef(filter.RequestMirror.BackendRef, "Spec.Rules.Filters.RequestMirror.BackendRef", KindGRPCRoute, route.Namespace)
+					if cond != nil {
+						routeAccessor.AddCondition(gatewayapi_v1beta1.RouteConditionType(cond.Type), cond.Status, gatewayapi_v1beta1.RouteConditionReason(cond.Reason), cond.Message)
+						continue
```

Comment:
Let's try to add a unit test for this code branch, in internal/dag/status_test.go